pydatastructs/graphs/algorithms.py

```diff
@@ -548,3 +550,57 @@ def strongly_connected_components(graph, algorithm):
         "isn't implemented for finding strongly connected components."
         %(algorithm, graph._impl))
     return getattr(algorithms, func)(graph)
+
+
+def random_walk(graph, start, path_length, seed = 0):
+    """
+    Executes random walk from the starting node and given graph.
+
+    Parameters
+    ==========
+
+    graph: Graph
+        The graph on which random walk needs to be executed.
+    start: GraphNode or it's subclasses
+        The node/vertex from where random walk will begin.
+    path_length:
+        The maximum number of nodes/vertices traversed by random walk.
+
+    Returns
+    =======
+
+    path: list
+        Python list of nodes after execution of random walk.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import Graph, AdjacencyListGraphNode
+    >>> from pydatastructs import random_walk
+    >>> v1, v2, v3, v4, v5, v6, v7, v8 = [AdjacencyListGraphNode(i) for i in range(8)]
+    >>> g = Graph(v1, v2, v3, v4, v5, v6, v7, v8)
+    >>> g.add_edge(v1.name, v2.name)
+    >>> g.add_edge(v1.name, v3.name)
+    >>> g.add_edge(v1.name, v4.name)
+    >>> g.add_edge(v2.name, v5.name)
+    >>> g.add_edge(v2.name, v6.name)
+    >>> g.add_edge(v3.name, v6.name)
+    >>> g.add_edge(v3.name, v7.name)
+    >>> g.add_edge(v4.name, v7.name)
+    >>> g.add_edge(v4.name, v8.name)
+
+    >>> random_walk = random_walk(g, v1, 3, 44)
+    >>> random_walk == ['0', '2', '6']
+    True
```

Comment:
Use some simple example. A three node graph(v1->v2->v3->v1) would be better. It should say that nodes can be repeated in a random walk.